arch/cpu/arm/cortex-m/cm4/Makefile.cm4

```diff
@@ -9,20 +9,10 @@ LDFLAGS += -Wl,-Map=$(CONTIKI_NG_PROJECT_MAP),--cref,--no-warn-mismatch
 
 OBJCOPY_FLAGS += --gap-fill 0xff
 
-### Build syscalls for newlib
-MODULES += os/lib/newlib
```

Comment:
Why remove this?

Reply:
I was getting link errors that I couldn't really be bothered with debugging :)

Reply:
```
$ make
using saved target 'nrf52dk'
SDK: /Users/cexgo/nrf-sdk
SoftDevice: /Users/cexgo/nrf-sdk/components/softdevice/s1xx_iot/s1xx-iot-prototype3_nrf52_softdevice.hex
  CC        ../../../../../os/lib/newlib/syscalls.c
  CC        ../resources/res-leds.c
  CC        coap-server.c
  LD        coap-server.elf
obj_nrf52dk/syscalls.o: In function `_sbrk':
/Users/cexgo/contiki-ng/examples/platform-specific/nrf52dk/coap-demo/coap-server/../../../../../os/lib/newlib/syscalls.c:82: undefined reference to `_eheap'
obj_nrf52dk/syscalls.o:(.data.heap_end.4576+0x0): undefined reference to `_heap'
collect2: error: ld returned 1 exit status
make: *** [coap-server.elf] Error 1
rm coap-server.o
```

Reply:
Any clues? Does not happen with all examples...

Reply:
Yes, actually. When you look at the original PR https://github.com/contiki-os/contiki/pull/1088, the author also added `_heap` and `_eheap` symbols for cc2538, but not for other CPU. So, these could be added to the linker scripts in `arch/cpu/nrf52832/ld/` (and also maybe for CC26xx?).

Reply:
Thanks Atis. Alternatively, maybe  we change _heap and _eheap to symbols that stock linker scripts actually provide?

Reply:
@atiselsts I've updated this pull, put the MODULE back in for all cortexes and adjusted linker scripts where required.